File: Funcional/localize.hs
import Data.List
import System.IO
import System.Directory
 
-- //////////////////////////////////////  DATA  //////////////////////////////////////
data City = City {
    nomeCidade :: String
} deriving (Show)

data Store = Store {
    nomeStore :: String,
    carros :: [Car],
    cidade :: City
} deriving (Show)

data Car = Car {
    codigo :: String,
    modelo :: String,
    quantidade :: Int,
    capacidade :: Int,
    diaria :: Float,
    categoria :: String
} deriving (Show)

data Rent = Rent {
    codigoRent :: String,
    veiculo :: Car,
    cidade_destino :: City,
    diarias :: Float,
    dias :: Int
    } deriving (Show)

data Client = Client {
    codigoCliente :: String,
    nomeClient :: String,
    cpf :: String,
    cnh :: String,
    alugueis :: [Rent]
    } deriving (Show)

-- //////////////////////////////////////  CADASTRO INICIAL  //////////////////////////////////////
cidade1 = City {nomeCidade = "João Pessoa"}
cidade2 = City {nomeCidade = "Campina Grande"}
cidade3 = City {nomeCidade = "Cajazeiras"}

carro1 = Car {codigo = "1", modelo = "BMW", quantidade = 10, capacidade = 5, diaria = 500.00, categoria = "luxo"}
carro2 = Car {codigo = "2", modelo = "BMW2", quantidade = 10, capacidade = 2, diaria = 600.00, categoria = "luxo"}
carro3 = Car {codigo = "3", modelo = "BMW3", quantidade = 10, capacidade = 5, diaria = 300.00, categoria = "normal"}
carro4 = Car {codigo = "4", modelo = "BMW4", quantidade = 10, capacidade = 2, diaria = 200.00, categoria = "normal"}
carro5 = Car {codigo = "5", modelo = "BMW5", quantidade = 10, capacidade = 5, diaria = 100.00, categoria = "normal"}

carrosCadastrados = [carro1, carro2, carro3, carro4, carro5]
 
loja1 = Store {nomeStore = "Loja A", carros = carrosCadastrados, cidade = cidade1}
loja2 = Store {nomeStore = "Loja B", carros = carrosCadastrados, cidade = cidade1}
loja3 = Store {nomeStore = "Loja C", carros = carrosCadastrados, cidade = cidade2}
loja4 = Store {nomeStore = "Loja D", carros = carrosCadastrados, cidade = cidade2}
loja5 = Store {nomeStore = "Loja E", carros = carrosCadastrados, cidade = cidade3}
 
cliente1 = Client {codigoCliente = "1", nomeClient = "Jesus", cpf = "111.111.111-25", cnh = "1000", alugueis = []}
cliente2 = Client {codigoCliente = "2", nomeClient = "Deus", cpf = "222.111.111-25", cnh = "2000", alugueis = []}
cliente3 = Client {codigoCliente = "3", nomeClient = "Jafeh", cpf = "333.111.111-25", cnh = "3000", alugueis = []}
cliente4 = Client {codigoCliente = "4", nomeClient = "Judas", cpf = "444.111.111-25", cnh = "4000", alugueis = []}
cliente5 = Client {codigoCliente = "5", nomeClient = "Maria", cpf = "555.111.111-25", cnh = "5000", alugueis = []}
 
clientesCadastrados = [cliente1, cliente2, cliente3, cliente4, cliente5]
 
-- //////////////////////////////////////  MENU  //////////////////////////////////////
 
main :: IO ()
main = do
   menuPrint
   menuOpcao


menuPrint :: IO ()
menuPrint = do
   putStrLn "██████╗ ███████╗███╗   ███╗    ██╗   ██╗██╗███╗   ██╗██████╗  ██████╗ "
   putStrLn "██╔══██╗██╔════╝████╗ ████║    ██║   ██║██║████╗  ██║██╔══██╗██╔═══██╗"
   putStrLn "██████╔╝█████╗  ██╔████╔██║    ██║   ██║██║██╔██╗ ██║██║  ██║██║   ██║"
   putStrLn "██╔══██╗██╔══╝  ██║╚██╔╝██║    ╚██╗ ██╔╝██║██║╚██╗██║██║  ██║██║   ██║"
   putStrLn "██████╔╝███████╗██║ ╚═╝ ██║     ╚████╔╝ ██║██║ ╚████║██████╔╝╚██████╔╝"
   putStrLn "╚═════╝ ╚══════╝╚═╝     ╚═╝      ╚═══╝  ╚═╝╚═╝  ╚═══╝╚═════╝  ╚═════╝ "


menuOpcao :: IO ()
menuOpcao = do
   putStrLn "À LOCALIZE CAR"
   putStrLn "O que você deseja fazer?"
   putStrLn "(1) Pesquisar/Alugar"
   putStrLn "(2) Devolver"
   putStrLn "(3) Sair"
   putStrLn "\nOpcao: "
   opcao <- getLine
   if (read opcao) == 3 then putStrLn("Fim") else do opcaoEscolhida (read opcao)

-- //////////////////////////////////////  ESCOLHER OPÇÃO  //////////////////////////////////////

opcaoEscolhida :: Int -> IO()
opcaoEscolhida opcao
   | opcao == 2 = do {pesquisar}
   | opcao == 3 = do {devolver}
   | otherwise =  do {putStrLn "Opcao invalida, por favor escolha uma opcao valida" ; menuOpcao}

pesquisar :: IO()
pesquisar = putStrLn ("\n\n\n" ++ "pesquisando...." ++ "\n\n")

devolver :: IO()
devolver = putStrLn ("\n\n\n" ++ "devolvendo...." ++ "\n\n")

File: PROJETO - PROLOG/localize.pl
:- initialization(main).

/*=========================== Auxiliares===============================*/
%Store(nome, [String]).
store("Localiza", cidadesTodas).
store("Movida", cidadesMovida).
store("Movida", cidadesMovida).
store("Hertz", cidadesHertz).

%Car( cod, model, amount, capacity, valorDiaria, category, store, localCity).
car("1", "BMW", 10, 5, 500.00, "luxo", "Loacalize", "João Pessoa").
car("2", "BMW2", 5, 2, 600.00, "luxo", "Loacalize", "João Pessoa").
car("3", "BMW3", 2, 10, 300.00, "normal", "Loacalize", "João Pessoa").
car("4", "BMW", 10, 5, 500.00, "luxo", "Loacalize", "Campina Grande").
car("5", "BMW2", 5, 2, 600.00, "luxo", "Loacalize", "Campina Grande").
car("6", "BMW3", 2, 10, 300.00, "normal", "Loacalize", "Campina Grande").
car("7", "BMW", 10, 5, 500.00, "luxo", "Loacalize", "Cajazeiras").
car("8", "BMW2", 5, 2, 600.00, "luxo", "Loacalize", "Cajazeiras").
car("9", "BMW3", 2, 10, 300.00, "normal", "Loacalize", "Cajazeiras").

%Rent(Car, destinationCity, diary, days).

%Client(cod, name, cpf, cnh, amountRent, [Rent]).
client("1", "Jesus", "111.111.111-25", "1000", 0, []).
client("2", "Deus", "222.111.111-25", "2000", 0, []).
client("3", "Jafeh", "333.111.111-25", "3000", 0, []).
client("4", "Judas", "444.111.111-25", "4000", 0, []).
client("5", "Maria", "555.111.111-25", "5000", 0, []).
	
menu() :- 
	writeln("Bem Vindo!").
	 	  	
opcao(1) :-	
	writeln("==> Digite a CIDADE DE PARTIDA: "), read(CidadeP),
	writeln("==> Digite a CIDADE DE DESTINO: "), read(CidadeD),
	writeln("==> Digite a QUANTIDADE DE PESSOAS: "), read(Qtdpessoas),
	imprimeCarros().
			
opcao(2) :- writeln("VAMOS DEVOLVER O CARRO"),
			write("Devolver").

opcao(3) :- halt.

opcao(_) :- writeln("Opcao invalida, tente outra!").
				
menuOpcoes() :-  	
	writeln(" "),
	writeln("1 - Pesquisar/Alugar"),
	writeln("2 - Devolver"),
	writeln("3 - Sair"),
	writeln("\nOpcao: "),
	read(A),
	opcao(A),
	menuOpcoes().
		
main :-
	menu(),	
	menuOpcoes().

listarCarros([]).
	listarCarros([Head|Tail]) :-
	car(Head,Nome,Qtd,Cap,Dia, Cat, Loja, Cidade),
	write(Head),
	write(" - "),
	write(Nome),
	write(" - "),
	write(Qtd),
	write(" - "),
	write(Cap),
	write(" - "),
	write(Dia),
	write(" - "),
	write(Cat),
	write(" - "),
	write(Loja),
	write(" - "),
	writeln(Cidade),
	 
	listarCarros(Tail).
	
imprimeCarros() :-
	writeln( "Código | Modelo | Capacidade | Quantidade | Diária | Categoria | Loja"),
	findall(Codigo, car(Codigo,_,_,_,_,_,_,_), Carros),
	listarCarros(Carros), menuAlugar. 

opcaoAlugar(1) :-
	opcao(1).
opcaoAlugar(2) :-
	writeln("Alugando...").
opcaoAlugar(3) :-
	halt.
opcaoAlugar(_) :-
	writeln("Opcao invalida, tente outra!"),
	menuAlugar.
	
menuAlugar :-
	writeln("O que deseja fazer? "),
	writeln("1 - Nova pesquisa"),
	writeln("2 - Alugar"),
	writeln("3 - Sair"),
	writeln("\nOpcao: "),
	read(A),
	opcaoAlugar(A).	
